docs/core/logging.md

```diff
@@ -200,27 +200,29 @@ You can also explicitly set a service name via **`POWERTOOLS_SERVICE_NAME`** env
+| **level**               | String  | "INFO"                                                                                               | Logging level                                                            |
+| **coldStart**           | Boolean | true                                                                                                 | ColdStart value.                                                         |
+| **service**             | String  | "payment"                                                                                            | Service name defined. "service_undefined" will be used if unknown        |
+| **samplingRate**        | int     | 0.1                                                                                                  | Debug logging sampling rate in percentage e.g. 10% in this case          |
+| **message**             | String  | "Collecting payment"                                                                                 | Log statement value. Unserializable JSON values will be casted to string |
+| **functionName**        | String  | "example-powertools-HelloWorldFunction-1P1Z6B39FLU73"                                                |                                                                          |
+| **functionVersion**     | String  | "12"                                                                                                 |                                                                          |
+| **functionMemorySize**  | String  | "128"                                                                                                |                                                                          |
+| **functionArn**         | String  | "arn:aws:lambda:eu-west-1:012345678910:function:example-powertools-HelloWorldFunction-1P1Z6B39FLU73" |                                                                          |
+| **xray_trace_id**       | String  | "1-5759e988-bd862e3fe1be46a994272793"                                                                | X-Ray Trace ID when Lambda function has enabled Tracing                  |
+| **function_request_id** | String  | "899856cb-83d1-40d7-8611-9e78f15f32f4""                                                              | AWS Request ID from lambda context                                       |
 
 ## Capturing context Lambda info
 
 You can enrich your structured logs with key Lambda context information via `logEvent` annotation parameter. 
 You can also explicitly log any incoming event using `logEvent` param. Refer [Override default object mapper](#override-default-object-mapper) 
 to customise what is logged.
 
+Note that the `logEvent` will work only if `POWERTOOLS_LOGGER_SAMPLE_RATE` environment variable is set to `"true"`
```

Comment:
is it the python behaviour ?

I think it is either one or the other to have the event logged, not both

Reply:
https://docs.powertools.aws.dev/lambda/python/latest/core/logger/#logging-incoming-event

Reply:
This is the TypeScript behavior:
https://github.com/aws-powertools/powertools-lambda-typescript/blob/aa2673d89a6676f4d9242773543876464b3b65ef/packages/logger/src/Logger.ts#L904

Reply:
We should follow python. It's either one or the other, objective is to simplify job, users choose either programatic way or with env var, we don't want to complexify and require both. More over, if we require both, it becomes a breaking change and should go in v2.